### PR TITLE
Separate docker initialisation from images

### PR DESCRIPTION
**Pull request by kitounliu**

### CODE REVIEW

File: ops/docker/Dockerfile.init [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM node:20
```
Comment (jyellick) [inline ops/docker/Dockerfile.init:1]: Why `node:20`? Seems like we should possibly be simply pulling a Cargo build env?  If we need Node at runtime for the ts scripts, perhaps we would be better off copying the built binary into a Node env? Presumably we do not need to build any rust at runtime?

File: services/admin.ts [hunk 2]
```diff
@@ -277,10 +299,10 @@ export class AdminZkRandService extends BaseService<AdminZkRandOptions> {
         await waitForWriteJsonToFile(obj, instancesPath)
         console.log("retrieved all instances from contract")
         console.log("sleeping..")
-        await sleep(2000)
+        await sleep(5000)
```
Comment (jyellick) [inline services/admin.ts:302]: Can we not wait for this asynchronously?

File: services/node.ts [hunk 3]
```diff
@@ -169,10 +192,10 @@ export class NodeZkRandService extends BaseService<NodeZkRandOptions> {
             const obj = JSON.stringify(ppListHex)
             await waitForWriteJsonToFile(obj, instancesPath)
             console.log("sleeping..")
-            await sleep(1000)
+            await sleep(5000)
```
Comment (jyellick) [inline services/node.ts:195]: Same comment here with respect to async wait?

**Review by jyellick — APPROVED** [review]
Overall, LGTM, a few things we might want to address, but nothing to hold the PR up

File: services/admin.ts (hunk 2 above)
Comment (kitounliu) [reply services/admin.ts:302]: writing to disk takes some time to finish. The command after will fail if no wait or not enough delay

File: ops/docker/Dockerfile.init (hunk 1 above)
Comment (kitounliu) [reply ops/docker/Dockerfile.init:1]: good point! will change that

**Pull request merged**